src/api/app/components/card_component.html.haml

```diff
@@ -0,0 +1,15 @@
+.col-12.col-md-6.col-lg-4
+  .card.mb-3.repository-card
+    .card-header.d-flex.justify-content-between
+      %div
+        = header
+      - if delete_button?
+        %div
+          = delete_button
```

Comment:
With this approach, we will have to duplicate the delete button over and over again. This could be streamlined to ensure the delete buttons are always consistent. This could be a component.

Reply:
That's the plan, to have a root component to reuse and do not _reinvent the wheel_ every time. I felt it was for a separate refactoring PR (as we do already have duplication around about it). Would you feel comfortable to keep it as it is for now, or do you prefer to add it here and now before merging the card component?

Reply:
Handling this where the card is used, that would be a good first step. It doesn't have to happen in this PR though.